### ISSUE TITLE
Dialogflow Gem issue in ruby:3.0.2-alpine Error loading shared library ld-linux-x86-64.so

### ISSUE THREAD

**Issue by sojan-official** [googleapis/google-cloud-ruby#13306]
#### Environment details

  - OS: Alpine
  - Ruby version: 3.0.2
  - Environment : Docker
  - Image : ruby:3.0.2-alpine
  - Gem name and version: google-cloud-dialogflow-v2 , (0.6.4)

#### Steps to reproduce

  1. Build a docker image using ruby:3.02-alpine. ref: https://github.com/chatwoot/chatwoot/blob/develop/docker/Dockerfile
  2. Accessing the gem methods from the rails app inside the image throw the following error 
  
  ```
LoadError: Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by /gems/ruby/3.0.0/gems/grpc-1.38.0-x86_64-linux/src/ruby/lib/grpc/3.0/grpc_c.so) - /gems/ruby/3.0.0/gem...
  ```

#### Code example

ref: https://github.com/chatwoot/chatwoot/blob/f477c5281f3049a15a19b9e7843bb8a578c02b8c/lib/integrations/dialogflow/processor_service.rb#L33

```ruby
Google::Cloud::Dialogflow.configure { |config| config.credentials = hook.settings['credentials'] }
session_client = Google::Cloud::Dialogflow.sessions
```

#### Full backtrace

```sh
2021-08-05T18:56:58.915151737Z 2021-08-05T18:56:58.914Z pid=1 tid=27c5 WARN: {"context":"Job raised exception","job":{"retry":true,"queue":"integrations","class":"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper","wrapped":"HookJob","args":[{"job_class":"HookJob","job_id":"844f184d-be4e-4153-99e4-7ba3ef4c2a94","provider_job_id":null,"queue_name":"integrations","priority":null,"arguments":[{"_aj_globalid":"gid://chatwoot/Integrations::Hook/1"},"message.created",{"message":{"_aj_globalid":"gid://chatwoot/Message/4"},"_aj_ruby2_keywords":["message"]}],"executions":0,"exception_executions":{},"locale":"en","timezone":"UTC","enqueued_at":"2021-08-05T18:30:52Z"}],"jid":"882ff241bbbf4eb461250e94","created_at":1628188252.7170382,"enqueued_at":1628189818.8984845,"error_message":"Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by /gems/ruby/3.0.0/gems/grpc-1.38.0-x86_64-linux/src/ruby/lib/grpc/3.0/grpc_c.so) - /gems/ruby/3.0.0/gems/grpc-1.38.0-x86_64-linux/src/ruby/lib/grpc/3.0/grpc_c.so","error_class":"LoadError","failed_at":1628188252.9348848,"retry_count":5,"retried_at":1628189005.1687806},"jobstr":"{\"retry\":true,\"queue\":\"integrations\",\"class\":\"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper\",\"wrapped\":\"HookJob\",\"args\":[{\"job_class\":\"HookJob\",\"job_id\":\"844f184d-be4e-4153-99e4-7ba3ef4c2a94\",\"provider_job_id\":null,\"queue_name\":\"integrations\",\"priority\":null,\"arguments\":[{\"_aj_globalid\":\"gid://chatwoot/Integrations::Hook/1\"},\"message.created\",{\"message\":{\"_aj_globalid\":\"gid://chatwoot/Message/4\"},\"_aj_ruby2_keywords\":[\"message\"]}],\"executions\":0,\"exception_executions\":{},\"locale\":\"en\",\"timezone\":\"UTC\",\"enqueued_at\":\"2021-08-05T18:30:52Z\"}],\"jid\":\"882ff241bbbf4eb461250e94\",\"created_at\":1628188252.7170382,\"enqueued_at\":1628189818.8984845,\"error_message\":\"Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by /gems/ruby/3.0.0/gems/grpc-1.38.0-x86_64-linux/src/ruby/lib/grpc/3.0/grpc_c.so) - /gems/ruby/3.0.0/gems/grpc-1.38.0-x86_64-linux/src/ruby/lib/grpc/3.0/grpc_c.so\",\"error_class\":\"LoadError\",\"failed_at\":1628188252.9348848,\"retry_count\":5,\"retried_at\":1628189005.1687806}"}
2021-08-05T18:56:58.915446269Z 2021-08-05T18:56:58.915Z pid=1 tid=27c5 WARN: LoadError: Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by /gems/ruby/3.0.0/gems/grpc-1.38.0-x86_64-linux/src/ruby/lib/grpc/3.0/grpc_c.so) - /gems/ruby/3.0.0/gems/grpc-1.38.0-x86_64-linux/src/ruby/lib/grpc/3.0/grpc_c.so
2021-08-05T18:56:58.915559093Z 2021-08-05T18:56:58.915Z pid=1 tid=27c5 WARN: /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
2021-08-05T18:56:58.915566668Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
2021-08-05T18:56:58.915574764Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
2021-08-05T18:56:58.915579042Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
2021-08-05T18:56:58.915583320Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
2021-08-05T18:56:58.915587437Z /gems/ruby/3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
2021-08-05T18:56:58.915591455Z /gems/ruby/3.0.0/gems/grpc-1.38.0-x86_64-linux/src/ruby/lib/grpc/grpc.rb:20:in `<main>'
2021-08-05T18:56:58.915595803Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
2021-08-05T18:56:58.915599992Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
2021-08-05T18:56:58.915604200Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
2021-08-05T18:56:58.915608538Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
2021-08-05T18:56:58.915612656Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
2021-08-05T18:56:58.915639307Z /gems/ruby/3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
2021-08-05T18:56:58.915643625Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:53:in `require_relative'
2021-08-05T18:56:58.915647853Z /gems/ruby/3.0.0/gems/grpc-1.38.0-x86_64-linux/src/ruby/lib/grpc.rb:19:in `<main>'
2021-08-05T18:56:58.915651840Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
2021-08-05T18:56:58.915655998Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
2021-08-05T18:56:58.915660226Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
2021-08-05T18:56:58.915665176Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
2021-08-05T18:56:58.915681507Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
2021-08-05T18:56:58.915685745Z /gems/ruby/3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
2021-08-05T18:56:58.915689692Z /gems/ruby/3.0.0/gems/gapic-common-0.3.4/lib/gapic/grpc.rb:15:in `<main>'
2021-08-05T18:56:58.915693811Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
2021-08-05T18:56:58.915697939Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
2021-08-05T18:56:58.915720872Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
2021-08-05T18:56:58.915725030Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
2021-08-05T18:56:58.915729228Z /gems/ruby/3.0.0/gems/bootsnap-1.7.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
2021-08-05T18:56:58.915733246Z /gems/ruby/3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
2021-08-05T18:56:58.915737343Z /gems/ruby/3.0.0/gems/google-cloud-dialogflow-v2-0.6.4/lib/google/cloud/dialogflow/v2/sessions/client.rb:139:in `initialize'
2021-08-05T18:56:58.915741511Z /gems/ruby/3.0.0/gems/google-cloud-dialogflow-1.2.0/lib/google/cloud/dialogflow.rb:227:in `new'
2021-08-05T18:56:58.915745569Z /gems/ruby/3.0.0/gems/google-cloud-dialogflow-1.2.0/lib/google/cloud/dialogflow.rb:227:in `sessions'
2021-08-05T18:56:58.915749586Z /app/lib/integrations/dialogflow/processor_service.rb:35:in `get_dialogflow_response'
2021-08-05T18:56:58.915753664Z /app/lib/integrations/dialogflow/processor_service.rb:10:in `perform'
2021-08-05T18:56:58.915757572Z /app/app/jobs/hook_job.rb:27:in `process_dialogflow_integration'
2021-08-05T18:56:58.915761510Z /app/app/jobs/hook_job.rb:9:in `perform'
2021-08-05T18:56:58.915765387Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/execution.rb:48:in `block in perform_now'
2021-08-05T18:56:58.915769365Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:117:in `block in run_callbacks'
2021-08-05T18:56:58.915773462Z /gems/ruby/3.0.0/gems/sentry-rails-4.6.4/lib/sentry/rails/active_job.rb:9:in `block (2 levels) in included'
2021-08-05T18:56:58.915777590Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec'
2021-08-05T18:56:58.915849106Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
2021-08-05T18:56:58.915855379Z /gems/ruby/3.0.0/gems/i18n-1.8.10/lib/i18n.rb:314:in `with_locale'
2021-08-05T18:56:58.915859366Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/translation.rb:9:in `block (2 levels) in <module:Translation>'
2021-08-05T18:56:58.915863775Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec'
2021-08-05T18:56:58.915893051Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
2021-08-05T18:56:58.915897499Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/core_ext/time/zones.rb:66:in `use_zone'
2021-08-05T18:56:58.915901466Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/timezones.rb:9:in `block (2 levels) in <module:Timezones>'
2021-08-05T18:56:58.915905544Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec'
2021-08-05T18:56:58.915909561Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
2021-08-05T18:56:58.915913639Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/instrumentation.rb:21:in `block in instrument'
2021-08-05T18:56:58.915917698Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/notifications.rb:203:in `block in instrument'
2021-08-05T18:56:58.915921735Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
2021-08-05T18:56:58.915925763Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/notifications.rb:203:in `instrument'
2021-08-05T18:56:58.915929841Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/instrumentation.rb:31:in `instrument'
2021-08-05T18:56:58.915933798Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/instrumentation.rb:14:in `block (2 levels) in <module:Instrumentation>'
2021-08-05T18:56:58.915938126Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec'
2021-08-05T18:56:58.915942184Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
2021-08-05T18:56:58.915946211Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/logging.rb:22:in `block in tag_logger'
2021-08-05T18:56:58.915950180Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/tagged_logging.rb:99:in `block in tagged'
2021-08-05T18:56:58.915954167Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/tagged_logging.rb:37:in `tagged'
2021-08-05T18:56:58.915974756Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/tagged_logging.rb:99:in `tagged'
2021-08-05T18:56:58.915979244Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/logging.rb:22:in `tag_logger'
2021-08-05T18:56:58.915983283Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/logging.rb:15:in `block (2 levels) in <module:Logging>'
2021-08-05T18:56:58.915987230Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec'
2021-08-05T18:56:58.915991268Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
2021-08-05T18:56:58.915995345Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:137:in `run_callbacks'
2021-08-05T18:56:58.915999583Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/execution.rb:47:in `perform_now'
2021-08-05T18:56:58.916003611Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/execution.rb:25:in `block in execute'
2021-08-05T18:56:58.916058275Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:117:in `block in run_callbacks'
2021-08-05T18:56:58.916063595Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/railtie.rb:47:in `block (4 levels) in <class:Railtie>'
2021-08-05T18:56:58.916068004Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/execution_wrapper.rb:88:in `wrap'
2021-08-05T18:56:58.916072061Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/reloader.rb:72:in `block in wrap'
2021-08-05T18:56:58.916076099Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/execution_wrapper.rb:84:in `wrap'
2021-08-05T18:56:58.916080288Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/reloader.rb:71:in `wrap'
2021-08-05T18:56:58.916084295Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/railtie.rb:46:in `block (3 levels) in <class:Railtie>'
2021-08-05T18:56:58.916088513Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `instance_exec'
2021-08-05T18:56:58.916092541Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
2021-08-05T18:56:58.916096789Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/callbacks.rb:137:in `run_callbacks'
2021-08-05T18:56:58.916100856Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/execution.rb:23:in `execute'
2021-08-05T18:56:58.916104774Z /gems/ruby/3.0.0/gems/activejob-6.1.4/lib/active_job/queue_adapters/sidekiq_adapter.rb:42:in `perform'
2021-08-05T18:56:58.916108761Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:196:in `execute_job'
2021-08-05T18:56:58.916244491Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:164:in `block (2 levels) in process'
2021-08-05T18:56:58.916252936Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/middleware/chain.rb:138:in `block in invoke'
2021-08-05T18:56:58.916257114Z /gems/ruby/3.0.0/gems/sentry-sidekiq-4.6.4/lib/sentry/sidekiq/sentry_context_middleware.rb:23:in `call'
2021-08-05T18:56:58.916261282Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/middleware/chain.rb:140:in `block in invoke'
2021-08-05T18:56:58.916265219Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/middleware/chain.rb:143:in `invoke'
2021-08-05T18:56:58.916269268Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:163:in `block in process'
2021-08-05T18:56:58.916273296Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:136:in `block (6 levels) in dispatch'
2021-08-05T18:56:58.916277333Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/job_retry.rb:112:in `local'
2021-08-05T18:56:58.916281261Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:135:in `block (5 levels) in dispatch'
2021-08-05T18:56:58.916307511Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/rails.rb:14:in `block in call'
2021-08-05T18:56:58.916311268Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/execution_wrapper.rb:88:in `wrap'
2021-08-05T18:56:58.916315285Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/reloader.rb:72:in `block in wrap'
2021-08-05T18:56:58.916325645Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/execution_wrapper.rb:88:in `wrap'
2021-08-05T18:56:58.916329743Z /gems/ruby/3.0.0/gems/activesupport-6.1.4/lib/active_support/reloader.rb:71:in `wrap'
2021-08-05T18:56:58.916333711Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/rails.rb:13:in `call'
2021-08-05T18:56:58.916337648Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:131:in `block (4 levels) in dispatch'
2021-08-05T18:56:58.916341656Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:257:in `stats'
2021-08-05T18:56:58.916345703Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:126:in `block (3 levels) in dispatch'
2021-08-05T18:56:58.916349701Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/job_logger.rb:13:in `call'
2021-08-05T18:56:58.916353648Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:125:in `block (2 levels) in dispatch'
2021-08-05T18:56:58.916357696Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/job_retry.rb:79:in `global'
2021-08-05T18:56:58.916361613Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:124:in `block in dispatch'
2021-08-05T18:56:58.916365612Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/logger.rb:11:in `with'
2021-08-05T18:56:58.916386170Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/job_logger.rb:33:in `prepare'
2021-08-05T18:56:58.916390869Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:123:in `dispatch'
2021-08-05T18:56:58.916394937Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:162:in `process'
2021-08-05T18:56:58.916398945Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:78:in `process_one'
2021-08-05T18:56:58.916402973Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/processor.rb:68:in `run'
2021-08-05T18:56:58.916406900Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/util.rb:43:in `watchdog'
2021-08-05T18:56:58.916410898Z /gems/ruby/3.0.0/gems/sidekiq-6.2.1/lib/sidekiq/util.rb:52:in `block in safe_thread'
```

**Comment by dazuma** [comment]
TL;DR: `bundle config set --local force_ruby_platform true` in your Dockerfile before installing the bundle. You may also need to install C compilers/development tools in your image in order to install some of the gems.

Details:

Two of the dependencies of the dialogflow client, the [grpc gem](https://rubygems.org/gems/grpc) and the [google-protobuf gem](https://rubygems.org/gems/grpc), are normally distributed with precompiled C code. Unfortunately, the Linux version of the precompiled gem releases is compiled against glibc and thus is not compatible with Alpine Linux.

To get around this, you need to install the _source versions_ of those two gems, and have Rubygems compile their C code against your Alpine runtime on gem installation. `bundle config set --local force_ruby_platform true` sets the appropriate config to tell bundler to do this; run that command before installing your bundle. Alternatively, you can set the `BUNDLE_FORCE_RUBY_PLATFORM` environment variable. See the [bundle config documentation](https://bundler.io/v2.2/man/bundle-config.1.html) for more details.

As a caveat, this will build google-protobuf and grpc from C source when those gems are installed, and this may take a few minutes. You might consider preinstalling those gems in a base image and caching it. Additionally, this will also require you to have installed C compilers in your image. If that's an issue, consider using a multi-stage Dockerfile, installing the bundle in a first stage that includes development tools, and copying them into a final stage without the development tools.
